Unix/doc/diagnose-omi-problems.md

```diff
@@ -0,0 +1,293 @@
+`EventId` identifiers.
+
+### Enabling logging in OMI
+
+Currently, OMI requires logging set to the `INFO` level or higher in
+order to diagnose problems. Since OMI's
+`/etc/opt/omi/conf/omiserver.conf` file doesn't currently have the
+logging entry in the configuration file, add the following lines to
+`/etc/opt/conf/omiserver.conf`:
+
+```
+##
+## loglevel -- set the loggiing options for MI server
+##   Valid options are: ERROR, WARNING, INFO, DEBUG, VERBOSE (debug build)
+##   If commented out, then default value is: WARNING
+##
+loglevel = INFO
+```
+
+Logging is then controlled by commenting or uncommenting the
+`loglevel = INFO` line with a `#` at the beginning of the line.
+
+**Note:** After making changes to `/etc/opt/omi/conf/omiserver.conf`,
+you must restart omi service for the changes to take effect. This can
+be done with a command like: `sudo /opt/omi/bin/service_control
+restart`.
+
+**Note:** OMI log files can grow over time. To reduce the size of the
+log files and simplify diagnosis of problems, you may wish to delete
+existing log files before diagnosing your problem. This can be achieved
+with commands like:
+
+```
+sudo /opt/omi/bin/service_control stop
+sudo rm /var/opt/omi/log/*.log
+sudo /opt/omi/bin/service_control start
+```
+
+### Log file entries of interest
+
+In general, we need to see OMI activity. This activity would include:
+
+1. Request/Operation (1 message),
+2. Instance posted (0 or more messages), and
+3. Actual result (MI_RESULT_OK, MI_RESULT_FAILED, etc) (1 message)
+
+Furthermore, if a provider actually aborts, we should see a message
+indicating this.
+
+OMI has two log files of interest:
+
+1. `/var/opt/omi/log/omiserver.log` represents server or engine log
+   messages, and
+2. `/var/opt/omi/log/omiagent.*.log` represents agent messages
+   generated by provider operations.
+
+Be aware that the exact name of the `omiagent.*.log` file depends on
+the account that the agent is running under, which is determined by
+the agent's registration file and the account performing the OMI
+operation.
+
+A sample log entry for the request/operation is (found in omiagent log):
+
+```
+2017/11/21 15:44:01 [79169,79169] INFO: null(0): EventId=40039 Priority=INFO New request received: command=(EnumerateInstancesReq), namespace=(root/scx), class=(SCX_Agent)
+```
+
+Note that this is currently logged as an INFO message. Thus, you must
+have logging set to INFO or higher to see this message.
+
+Currently, the instances posted and the actual result don't log in an
+easy to retrieve mechanism. This will be resolved in a future OMI
+release. However, we do log the following message when a request is
+completely processed:
+
+```
+2017/11/21 15:44:02 [79169,79169] INFO: null(0): EventId=40011 Priority=INFO done with receiving msg(0x151d218:4099:EnumerateInstancesReq:3)
+```
+
+If you don't see an entry like the above (note that PID is shown in
+angle brackets), that likely indicates a crash or abnormal abort of
```

Comment:
Do't see angle brackets (<>).  Did you mean square brackets([])?

Reply:
Yup, fixed, thanks for catching this.